src/Form/index.tsx

```diff
@@ -624,20 +624,20 @@ function Form({
             asyncWrappedCode
           );
           try {
-            await fn(
+            const result = fn(
               { ...props, http: httpHelpers(client) },
               ...Object.values(injectableFields)
-            );
-          } catch (e) {
-            // rule had an error, log it to console for now
-            console.warn(
-              'Exception while running rule: ',
-              logicRule.name,
-              ' On Event: ',
-              logicRule.trigger_event,
-              ' Exception: ',
-              e
-            );
+              // );
+            ).catch((e: any) => {
+              // catch unhandled rejections in async user code (if a promise is returned)
+              // handle any errors in async code that actually returns a promise
+              handleRuleError(e.message, logicRule);
+            });
+
+            await result;
```

Comment:
why was the await pulled out to here?

Reply:
Yeah this is left-over from when I was trying several different things.  Let me put it back and retest that.